controllers/feeds/getOneFeedChat.js

```diff
@@ -83,6 +84,40 @@ const getOneFeedChatService = async (req, res) => {
     }
 
     const actor = data?.results[0]?.actor;
+
+    let karmaActors = [];
+    if (!karmaActors.includes(actor.id)) {
+      karmaActors.push(actor.id);
+    }
+    for (const comment of comments) {
+      if (!karmaActors.includes(comment?.reaction?.user_id)) {
+        karmaActors.push(comment?.reaction?.user_id);
+      }
+    }
+
+    const karmaScores = await UsersFunction.getUsersKarmaScore(User, karmaActors);
+    comments = comments.map((comment) => {
+      const user = karmaScores.find((user) => user.user_id === comment.reaction.user_id);
+      if (user && comment.reaction.user) {
+        comment.reaction.user.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+      }
+      if (comment.reaction.latest_children.comment) {
+        comment.reaction.latest_children.comment = comment.reaction?.latest_children?.comment?.map(
+          (child) => {
+            const user = karmaScores.find((user) => user.user_id === child.user_id);
+            if (user && child.user) {
+              child.user.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+              return child;
+            }
+          }
+        );
+      }
+      return comment;
+    });
+
+    const user = karmaScores.find((user) => user.user_id === actor.id);
+    actor.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code seems to be handling the retrieval and assignment of karma scores for actors and comments. However, there's a potential performance issue with the way unique user IDs are being collected into the `karmaActors` array. The `Array.includes()` method has a time complexity of O(n), which means that for each iteration over the comments, you're performing an operation that scales linearly with the number of elements in `karmaActors`. This could lead to performance issues when dealing with a large number of comments or users.

A more efficient approach would be to use a `Set` instead of an `Array` for `karmaActors`, as checking for existence in a set is generally faster (O(1) time complexity). Here's how you can modify your code:

```diff
-    let karmaActors = [];
+    let karmaActors = new Set();
-    if (!karmaActors.includes(actor.id)) {
+    if (!karmaActors.has(actor.id)) {
       karmaActors.add(actor.id);
     }
     for (const comment of comments) {
-      if (!karmaActors.includes(comment?.reaction?.user_id)) {
+      if (!karmaActors.has(comment?.reaction?.user_id)) {
         karmaActors.add(comment?.reaction?.user_id);
       }
     }
+    karmaActors = Array.from(karmaActors); // Convert back to array if necessary for getUsersKarmaScore function
```

Also, please ensure proper error handling for the asynchronous operation `UsersFunction.getUsersKarmaScore(User, karmaActors)`. If this operation fails, it could lead to unhandled promise rejection. Consider wrapping it in a try-catch block.


<!-- This is an auto-generated comment by OSS CodeRabbit -->